src/main/kotlin/com/beust/klaxon/Lookup.kt

```diff
@@ -1,12 +1,21 @@
 package com.beust.klaxon
 
+import java.math.BigInteger
 import java.util.ArrayList
 
 
 suppress("UNCHECKED_CAST")
 public fun <T> JsonObject.array(fieldName: String) : JsonArray<T>? = get(fieldName) as JsonArray<T>?
 public fun JsonObject.obj(fieldName: String) : JsonObject? = get(fieldName) as JsonObject?
-public fun JsonObject.long(fieldName: String) : Long? = get(fieldName) as Long?
+public fun JsonObject.int(fieldName: String) : Int? = get(fieldName) as Int?
+public fun JsonObject.long(fieldName: String) : Long? {
+    val value = get(fieldName)
+    when(value) {
```

Comment:
@fboldog so what unusual with it? it is normal behaviour as these types stored in memory in different ways